header/sync/sync.go

```diff
@@ -59,10 +58,6 @@ func NewSyncer(exchange header.Exchange, store header.Store, sub header.Subscrib
 
 // Start starts the syncing routine.
 func (s *Syncer) Start(context.Context) error {
-	if s.cancel != nil {
```

Comment:
Why remove?

Reply:
To avoid races. Please, check next comment.

Reply:
ack